public/js/account-linking-persistence.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Class for use with managing 
+ */
+
+class accountLinkingPersistence {
+  constructor() {
+    this._linked = false;
+    this._declined = false;
+    this.accessToken = undefined;
+    this.refresh();
```

Comment:
Why is this needed?

Reply:
I use `this.refresh()` in the constructor to attempt to pull down the latest state from localStorage. Theoretically the class could just do this as part of the getters, but I felt like having a record of state at instantiation would make more sense in this example, rather than querying the "store" each time a request came in.

Reply:
Got it, makes sense!